PollBuddy-Server/backend/modules/utils.js

```diff
@@ -40,6 +40,37 @@ async function validateID(collection, id) {
   return null;
 }
 
+// check if user is logged in. returns true or false.
+function isLoggedIn(req) {
+  return req.session.userData && req.session.userData.userID;
+}
+
+// Middleware to check if login is required for the poll.
+// poll ID must be present in req.params.id (in the url).
+// will attach a valid objectID on req.valid.id.
+async function checkPollPublic(req, res, next) {
+  // validate poll id
+  const id = await validateID("polls", req.params.id);
+  if (!id) {
+    return res.status(400).send(createResponse(null, "Invalid ID."));
+  }
+  req.parsedID = id;
+  // check poll publicity
+  try {
+    const poll = await mongoConnection.getDB().collection("polls").findOne({"_id": id});
```

Comment:
This should probably have a filter to only return that one property for efficiency.

Reply:
I think you just add `, { Public: true }` after the `{"_id": id}` but I'm not 100% certain.